src/components/common/Input/index.tsx

```diff
@@ -8,13 +8,15 @@ type InputProps = {
   register: UseFormRegister<FieldValues>;
   error: boolean;
   onChange?: () => void;
+  name: string;
+  placeholder: string;
 };
 
-const Input: FC<InputProps> = ({ label, type, register, error, onChange }) => (
+const Input: FC<InputProps> = ({ label, type, register, error, onChange, name, placeholder }) => (
   <div className="inputContainer">
-    <label className={`label__${label}`}>{label}</label>
-    <input type={type} className={`${error ? `error` : ``}`} id={label} {...register(label, { onChange })} />
+    <label className={`input-label input-label__${name}`}>{label}</label>
+    <input type={type} placeholder={placeholder} className={`input-field ${error ? `error` : ``}`} {...register(name, { onChange })} />
```

Comment:
`Classnames` library can help you here, with the class naming. 
You can take a look in the documentation and see if it fits your needs. It basically helps to avoid template strings and add conditionals easier.

https://jedwatson.github.io/classnames/